CODING-GUIDELINES.md

```diff
@@ -0,0 +1,163 @@
+- Structural directives like `*ngIf` and `*ngFor`
+- Inputs sorted alphabetically
+- Outputs sorted alphabetically
+- Other attributes like `class`, `tabIndex` etc. - sorted alphabetically
+
+### Whitespace between siblings
+
+If two elements are on the same level in the DOM hierarchy, put whitespace between these elements, in order to improve
+readability.
+
+#### **`test.component.html`**
+```angular2html
+<div>
+  <p>test</p>
+  <!-- whitespace -->
+  <p>test 2</p>
+</div>
+```
+
+## Typescript
+
+### Injecting services
+
+Inject services into components with `private readonly`:
+
+#### **`test.component.ts`**
+
+```typescript
+...
+
+constructor(private readonly testService: TestService) {
+
+}
+
+...
+```
+
+### Property access modifiers
+
+All properties and methods in a class should be marked with access modifiers.
+
+#### Private
+
+Any properties not used outside of a component, should be marked as private.
+The property name should be prefixed with an underscore: `private _testProperty: string;`.
+
+#### Readonly
+
+Any properties with a constant reference, should be marked as readonly. For example:
+`public readonly testObservable$ = new BehaviourSubject<string>('')`.
+
+#### Constants
+
+All constant properties should be written in screaming snake case: `public readonly MY_VALUE = 'my value';'`
+
+### Property typing
+
+Unless initialized with a primitive or an explicit constructor, all properties and methods should have a return type specified:
+
+#### **`test.component.ts`**
+
+```typescript
+...
+
+public myProperty!: TestObject;
+
+private _myString  = 'test';
+
+public myVoidFunction(testParams: TestParam): void {
+    ...
+}
+
+private myTestFunction(testParams: TestParam): TestResult {
+    let testResult!: TestResult;
+
+    ...
+
+    return testresult
+}
+
+
+...
+```
+
+### Preferred class property order
+
+As much possible, please keep to the following ordering of properties in Angular classes:
+
+- Property decorators like `ViewChild`, `ViewChildren` etc.
+- `Input()`
+- `Output()`
+- Public properties
+- Public readonly properties
+- Private properties
+- Private readonly property
+- Getters/setters
+- Constructor
+- Angular lifecycle hooks
+  - ngOnInit
+  - ngAfterViewInit
+  - ngOnChanges
+  - ngOnDestroy
+- Public methods
+- Private methods
+
+#### Sort properties by domain and alphabetically
+
+Properties and methods belonging to a certain logical domain should be bundled together within a class. The bundle
+itself should be sorted alphabetically. This is a recommended way of organizing a class, and can be deviated from if
+another way is more practical.
+
+We recommend that logical grouping of methods takes precedence over grouping by access modifiers, since it is more likely
+that a developer will want to have easy access to methods relating to a certain domain, rather than methods with the same
+access modifier. Sorting by access modifier takes precedence over alphabetical sorting.
```

Comment:
```suggestion
We recommend that logical grouping of methods takes precedence over grouping by access modifiers, since it is more likely that a developer will want to have easy access to methods relating to a certain domain, rather than methods with the same access modifier.
Sorting by access modifier takes precedence over alphabetical sorting.
```